agent/engine/docker_task_engine.go

```diff
@@ -825,8 +830,9 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 				task.Arn, container.Name, mderr)
 		}
 	}
-
-	metadata := client.CreateContainer(config, hostConfig, dockerContainerName, createContainerTimeout)
+	ctx, cancel := context.WithTimeout(engine.ctx, createContainerTimeout)
```

Comment:
This is not an optimal way of doing this. All of the responsibility of creating the child context for timing out should be happening within the docker_client methods. Please change all of  these methods so that context creation happens within docker_client methods. Just that the parent for those contexts need to flow from here.

Reply:
See comments above.